common/membership/hashring.go

```diff
@@ -207,12 +201,19 @@ func (r *ring) MemberCount() int {
 }
 
 func (r *ring) Members() []*HostInfo {
```

Comment:
Question: Why is this an array of pointers?

Reply:
We use `*HostInfo` in our code base everywhere, so this led to `[]*HostInfo`

I'm about to commit changes to get rid of all pointers to `Hostinfo`